flink-tests/src/test/java/org/apache/flink/test/runtime/NettyEpollITCase.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.test.runtime;
+
+import org.apache.flink.api.java.functions.KeySelector;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.test.util.MiniClusterResource;
+import org.apache.flink.test.util.MiniClusterResource.MiniClusterResourceConfiguration;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+import static org.apache.flink.runtime.io.network.netty.NettyConfig.TRANSPORT_TYPE;
+import static org.apache.flink.util.ExceptionUtils.findThrowableWithMessage;
+
+/**
+ * Test network stack with taskmanager.network.netty.transport set to epoll. This test car only run
+ * on linux. On other platforms it's basically a NO-OP. See
+ * https://github.com/apache/flink-shaded/issues/30
+ */
+public class NettyEpollITCase extends TestLogger {
+
+	private static final Logger LOG = LoggerFactory.getLogger(NettyEpollITCase.class);
+
+	private static final int TASK_MANAGERES = 2;
+
+	@Test
+	public void testNettyEpoll() throws Exception {
+		Optional<MiniClusterResource> cluster = trySetUp();
+		if (!cluster.isPresent()) {
+			return;
+		}
+
+		try {
+			StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+			env.setParallelism(TASK_MANAGERES);
+			env.getConfig().disableSysoutLogging();
+
+			DataStream<Integer> input = env.fromElements(1, 2, 3, 4, 1, 2, 3, 42);
+			input.keyBy(new KeySelector<Integer, Integer>() {
+					@Override
+					public Integer getKey(Integer value) throws Exception {
+						return value;
+					}
+				})
+				.sum(0)
+				.print();
+
+			env.execute();
```

Comment:
Is it strictly necessary that this test must be an integration test case? As far as I understand the test case, wouldn't it be enough to setup a `NettyServer` with `epoll` activated and sending some data to it via a `NettyClient` with `epoll` activated? Since we don't assert anything else than whether the Flink program executes or not, it should be basically the same.

Reply:
The same question would remain: whether such artificial setup, setups a network stack in the same way as Flink.